src/lib/Nbe.ml

```diff
@@ -381,9 +388,21 @@ struct
     | true -> ret con
     | _ -> do_rigid_coe abs r s con
 
+
+  and should_do_rigid_coe line r s con =
```

Comment:
This could be made more efficient and somehow fused with the other code to avoid duplicated evaluation, but let's not get ahead of ourselves until we know it works lol